tasks/main.yml

```diff
@@ -18,6 +18,13 @@
     - ansible_virtualization_type != "container"
   register: selinux_manage_selinux
 
+- name: restore selinux contexts
+  shell: |
+    restorecon="$( find /sbin /usr/sbin -iname restorecon )"
+    $restorecon -R /usr /etc /var /dev /bin /sbin
+  when:
+    - selinux_manage_selinux.changed
```

Comment:
This should be a handler, that helps to make the task simpler (you can remove the when).

Reply:
Handlers are called after tasks.... so if we switch the above to a handler, the reboot task will be ran *before* selinux contexts are restored.  This leaves us in the wedged state we were trying to avoid. :)

Reply:
Okay, in that case I still suggest to use a handler and call `flush_handers` (in the [meta](https://docs.ansible.com/ansible/latest/modules/meta_module.html) module).

Having `when: changed` causes linting issue.

The restorecon command can be pinned in `vars/main.yml`, something like this:

```yaml
---
_selinux_restorecon:
  default: /usr/sbin/restorecon
  Debian: /sbin/restorecon

selinux_restorecon: "{{ _selinux_restorecon[ansible_os_family] | default(_selinux_restorecon['default']) }}"
```

That saves a step in the tasklist and [keeps it simple to read](https://robertdebock.nl/presentations/perfect-ansible/#/4).

And, where does the the list of partitions (`/usr /etc /var /dev /bin /sbin`) come from?

Reply:
Hmmm... I didn't know about the ```flush_handlers``` command.  That is useful.

However I'm not sure that is still the right thing to do.  ```flush_handlers``` will flush **all** pending handlers.  If this role is being ran with other roles there could be other pending handlers that one may or may not want to be triggered at that point in time.  e.g. maybe there is pending reboot from another role and if that is triggered before the restore selinux context ran.... well, now you're locked out(again).

/shrug  maybe a comment to what is happening & why is the solution to keeping things simple to read?

Maybe use ```skip_ansible_lint``` to avoid the linting noise for the step?

Didn't know about pinning commands, that's neat.  In that form will restorecon always be ran?  What if the target is a container and/or doesn't have ```restorecon``` installed? 

The list of partitions just came from experience of where missing contexts breaks things.  Thought about adding ```/home``` but missing contexts there don't usually break things.